src/test/app/Discrepancy_test.cpp

```diff
@@ -0,0 +1,4089 @@
+  Copyright (c) 2012-2016 Ripple Labs Inc.
+
+  Permission to use, copy, modify, and/or distribute this software for any
+  purpose  with  or without fee is hereby granted, provided that the above
+  copyright notice and this permission notice appear in all copies.
+
+  THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+  WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+  MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+  ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+  WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+  ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/test/jtx.h>
+#include <ripple/test/jtx/Env.h>
+#include <ripple/beast/unit_test.h>
+#include <ripple/beast/core/LexicalCast.h>
+#include <ripple/protocol/JsonFields.h>
+#include <ripple/protocol/SField.h>
+#include <boost/filesystem.hpp>
+#include <ripple/basics/StringUtilities.h>
+
+namespace ripple {
+
+extern std::array<char const*,20> ledgerXRPDiscrepancyData;
+
+class Discrepancy_test : public beast::unit_test::suite
+{
+    // This is a legacy test ported from js/coffee. It starts with
+    // a saved ledger, submits a transaction as a blob and then queries
+    // that transaction to verify that the net of balance changes match
+    // the fee charged.
+    void
+    testXRPDiscrepancy ()
+    {
+        testcase ("Discrepancy test : XRP Discrepancy");
+        using namespace test::jtx;
+
+        boost::system::error_code ec;
+        // create a temporary path to write DB files to and
+        // our starting ledger data
+        auto path  = boost::filesystem::temp_directory_path(ec);
+        BEAST_EXPECTS(!ec, ec.message());
+        path /= boost::filesystem::unique_path("%%%%-%%%%-%%%%-%%%%", ec);
+        BEAST_EXPECTS(!ec, ec.message());
+        boost::filesystem::create_directories(path, ec);
+        BEAST_EXPECTS(!ec, ec.message());
+
+        auto ledgerFile = path / "ledgerdata.json";
+        std::ofstream o (ledgerFile.string(), std::ios::out | std::ios::trunc);
+        for(auto p : ledgerXRPDiscrepancyData)
+            o << p;
+        o.close();
+
+        {
+            Env env(*this, [&]()
+                {
+                    auto p = std::make_unique<Config>();
+                    test::setupConfigForUnitTests(*p);
+                    p->START_LEDGER = ledgerFile.string();
+                    p->START_UP = Config::LOAD_FILE;
+                    p->legacy("database_path", path.string());
+                    return p;
+                }());
+
+
+            Json::Value jrq;
+            jrq[jss::id] = 1;
+            jrq[jss::tx_blob] =
```

Comment:
This probably originally came as a blob in the npm test.  A blob has the advantage that it's hard to mess with.  However it has the disadvantage that it's opaque.  It's not possible to figure out what the transaction is doing by examining the contents in a text editor.

In my opinion the test would be slightly more maintainable if the submitted transaction were in human-readable form, rather than a blob.  But it's probably not worth the effort of doing the conversion.